dynamodb/src/main/scala/zio/dynamodb/DynamoDBQuery.scala

```diff
@@ -332,8 +342,9 @@ object DynamoDBQuery {
   import scala.collection.immutable.{ Map => ScalaMap }
   import scala.collection.immutable.{ Set => ScalaSet }
 
-  sealed trait Constructor[+A] extends DynamoDBQuery[A]
-  sealed trait Write[+A]       extends Constructor[A]
+  sealed trait Constructor[+A]   extends DynamoDBQuery[A]
+  sealed trait TransactWrite[+A] extends Constructor[A]
+  sealed trait Write[+A]         extends TransactWrite[A]
 
```

Comment:
I think the only issue with this is that we would have to either use different write case classes for transactions vs batches or we would have to have the write case class extend both `TransactWrite` and `Write` so that it could be used in both batches and transactions. I can ask John what his thoughts are on this the next time I meet with him.

Reply:
So based on my conversation with John he suggested avoiding this kind of trait change because it's object-oriented in style. Instead he suggested writing something like `def transaction[A](query: DyanmoDBQuery[A]): DyanmoDBQuery[A]` that would convert the query into a transaction. If there are parts of the query that are not transactable then we can fail and produce an error message with the operator that is invalid.